TappNetwork/home-visiting-app-web
# Cu 3nnbz8r messaging overhaul

# Details
This PR replaces the Referral and shares system with a facebook messenger clone. Users can freely message each other and share forms and internal resources within the messenger.

# Step to Test
1. Pull down and run migrations/seeder in backend repo
2. Login with admin@dev and secret
3. Go to messages inbox and test search functionality
4. clear search query and see 'New Message' tags
5. click on thread
6. go back to index and make sure 'New Message' tag disappears 
7. click on thread and send a message
8. go back to inbox and make sure thread is moved to top of page and new message is present
9. use search feature and click on message
10. make sure searched for message is scrolled to in messenger applet 
11. go to resource center, click share button on resource, complete modal
12. go to inbox, and make sure share was successful
13. repeat for Forms

# Issues Addressed
[3nnbz8r](https://app.clickup.com/t/3nnbz8r)

# Screenshots
<img width="781" alt="Screen Shot 2022-10-27 at 2 40 52 PM" src="https://user-images.githubusercontent.com/29612767/198393823-106526d2-75af-4d39-aa6f-4527cf46151d.png">
<img width="509" alt="Screen Shot 2022-10-27 at 2 42 13 PM" src="https://user-images.githubusercontent.com/29612767/198393842-a4ca4c32-f362-4e1b-abee-af57f80621f8.png">
<img width="512" alt="Screen Shot 2022-10-27 at 2 41 13 PM" src="https://user-images.githubusercontent.com/29612767/198393858-8f9d6013-ee0a-464c-abea-3fcce7ea8514.png">
<img width="986" alt="Screen Shot 2022-10-27 at 2 41 04 PM" src="https://user-images.githubusercontent.com/29612767/198393865-85cf24ef-ce62-4ef0-aa5d-b7432d6b29a5.png">
